python-backend/wolfram_client.py

```diff
@@ -36,8 +41,17 @@ def ask(query: str, include_pod: str = None) -> str:
             logger.error("Wolfram API returned error", e)
         else:
             try:
-                return json.loads(result.content)
-            except Exception as e:
+                result = json.loads(result.content)
+                try:
+                    result["queryresult"]["error"]["code"]
+                except (AttributeError, TypeError):
+                    # if the key does not exist or is false then the api call was successful
+                    pass
+                else:
+                    raise APIError(result["queryresult"]["error"]["msg"])
```

Comment:
I had to lookup this syntax :)

Can this statement be moved to follow the statement on line 46 to achieve the same behavior?

Reply:
well, apparently it can!